location/src/dbus.rs

```diff
@@ -0,0 +1,38 @@
+use eyre::Result;
+use orb_backend_status_dbus::{BackendStatusProxy, WifiNetwork};
+use zbus::Connection;
+
+use crate::data::NetworkInfo;
+
+pub struct BackendStatus {
+    backend_status_proxy: BackendStatusProxy<'static>,
+}
+
+impl BackendStatus {
+    pub async fn new() -> Result<Self> {
+        let connection = Connection::session().await?;
```

Comment:
very important that the session can be passed in via dependency injection (i.e. provided as an argument, instead of constructed in the function). This allows the code to be more testable, and lets us set up a temporary dbus in tests, constructing the connection from it, and then passing that connection to the code. Without that, there is no way to independently test code, since it will all be sharing the same bus.

For an example of how this works, see [here](https://github.com/worldcoin/orb-software/blob/main/supervisor/tests/it/main.rs) and [here](https://github.com/worldcoin/orb-software/blob/main/supervisor/tests/it/helpers.rs)